framework/src/loops/ComputeMortarFunctor.C

```diff
@@ -428,32 +478,25 @@ ComputeMortarFunctor::projectQPoints3d(const Elem * msm_elem,
     if (current_iterate < max_iterates)
     {
       // Transfer quadrature point from sub-element to element and store
-      q_pts[qp] = transform_qp(xi(0).value(), xi(1).value());
+      q_pts.push_back(transform_qp(xi(0).value(), xi(1).value()));
 
       // The following checks if quadrature point falls in correct domain.
       // On small mortar segment elements with very distorted elements this can fail, instead of
       // erroring simply truncate quadrature point, these points typically have very small
       // contributions to integrals
       if (primal_elem->type() == TRI3 || primal_elem->type() == TRI6)
       {
-        if (q_pts[qp](0) < 0 || q_pts[qp](1) < 0 || q_pts[qp](0) + q_pts[qp](1) > 1)
+        if (q_pts.back()(0) < 0 || q_pts.back()(1) < 0 || q_pts.back()(0) + q_pts.back()(1) > 1)
         {
-          mooseWarning("Quadrature point: ", q_pts[qp], " out of bounds, truncating.");
-          q_pts[qp](0) = std::max(0., q_pts[qp](0));
-          q_pts[qp](1) = std::max(0., q_pts[qp](1));
-          if (q_pts[qp](0) + q_pts[qp](1) > 1)
-            q_pts[qp](0) = 1 - q_pts[qp](1);
+          mooseError("Quadrature point: ", q_pts.back(), " out of bounds, truncating.");
```

Comment:
The message says "truncating", but this is an outright let's-stop-the-run. Does it make sense to convert this into a `mooseException` in the hope that by doing so the elements will be less distorted?

Reply:
Thanks @recuero ! I agree `mooseException` seems to be better than `Error `or `Warning` here

Reply:
You're right, I actually don't think we need this anymore but an exception will work. During development I had a memory leak, I wasn't initializing some ADReal derivative values with 0 so occasionally the Newton iteration would struggle and I'd get points outside the element. This truncation was an attempted fix but later when I found the memory leak I ended up putting these errors back in because the projection should never fail. 

Anyway, either will probably work just fine but we should (in theory) never be running into this so up to you whether you want to keep it or not.